qemu/tests/live_backup_base.py

```diff
@@ -0,0 +1,248 @@
+
+    def transaction_add(self, args_list, type, data):
+        """
+        Generate a args list for a transaction.
+        :param args_list: the arg_list param that will be modified
+        :param type: type of the command
+        :param data: data of the command
+        """
+        args = {"type": type, "data": data}
+        args_list.append(args)
+
+    def backup_check(self, compare_image=None):
+        """
+        Check and compare the backup images with qemu-img
+        :param compare_image: the image that need to be compared
+        """
+        data_dir = self.data_dir
+        for image in self.image_chain:
+            params = self.params.object_params(image)
+            qemu_image = qemu_storage.QemuImg(params, data_dir, image)
+            qemu_image.check_image(params, data_dir)
+        params = self.params.object_params(self.source_image)
+        if compare_image:
+            compare_image = utils_misc.get_path(data_dir, compare_image)
+            source_image = qemu_storage.storage.get_image_filename(params, data_dir)
+            qemu_image.compare_images(compare_image, source_image)
+
+    def wait_for_complete(self):
+        """
+        Waiting for the block job completed, utils timeout; if still not get
+        complete event, raise TestFail exception;
+        """
+        monitor = self.vm.monitor
+        timeout = self.params.get("wait_timeout")
+        if monitor.protocol == "qmp":
+            monitor.clear_event("BLOCK_JOB_COMPLETED")
+        logging.info("Wait for block job completed...")
+        completed = utils_misc.wait_for(lambda: monitor.get_event("BLOCK_JOB_COMPLETED"),
+                                        first=3.0, step=3.0, timeout=timeout)
+        if not completed:
+            raise error.TestFail("Wait block job complete "
+                                 "timeout in %ss" % timeout)
+
+    def reopen(self):
+        """
+        Closing the vm and reboot it with the image given.
+        """
+        vm = self.vm
+        vm.destroy()
+        image_chain = self.image_chain
+        image_name = self.params.get("image_name_%s" %
+                                     image_chain[len(image_chain) - 1])
+        self.params["image_name_%s" % self.source_image] = image_name
+        vm.create(params=self.params)
+        self.vm = vm
+        session = vm.wait_for_login(
+            timeout=int(self.params.get("login_timeout", 360)))
+        session.close()
+
+    def get_md5sum(self, create=False):
+        """
+        Get md5sum value of a file.
+        :param create: create a new file if True
+        :return: md5sum value
+        """
+        params = self.params
+        session = self.vm.wait_for_login(
+            timeout=int(self.params.get("login_timeout", 360)))
+        file_create_cmd = params.get("create_cmd")
+        file_name = params.get("file_name")
+        md5sum_cmd = params.get("md5sum_cmd")
```

Comment:
1. Can use hash_file() in the framework.
2. IMO, it's better to implement such general func() in the block_copy / framework.